src/redis/driver_impl.go

```diff
@@ -133,7 +156,14 @@ func NewClientImpl(scope stats.Scope, useTls bool, auth, redisSocketType, redisT
 		if len(urls) < 2 {
 			panic(RedisError("Expected master name and a list of urls for the sentinels, in the format: <redis master name>,<sentinel1>,...,<sentineln>"))
 		}
-		client, err = radix.NewSentinel(urls[0], urls[1:], radix.SentinelPoolFunc(poolFunc))
+
+		sentConnOpt := radix.SentinelConnFunc(sdf)
+		opt := radix.SentinelPoolFunc(poolFunc)
```

Comment:
Thank you for this 🫡 

Chiming in since we had a fun day with Sentinel and ratelimit today and I came across this change.

This didn't work to fix all our Sentinel woes. The missing piece here is that `poolFunc` also needs the custom connection `sdf` for sentinel.

`SentinelConnFunc` only applies to the sentinel connection itself and the pool is what's used for primary and secondary connections